src/components/AddItem.js

```diff
@@ -1,31 +1,124 @@
 import React from 'react';
 import { useState } from 'react';
 import { db } from '../lib/firebase';
-import { collection, addDoc } from '@firebase/firestore';
+import {
+  collection,
+  doc,
+  getDocs,
+  updateDoc,
+  arrayUnion,
+  setDoc,
+} from '@firebase/firestore';
 
 const AddItem = () => {
-  const [loading, setLoading] = useState(false);
-  const [error, setError] = useState(false);
+  const [itemName, setItemName] = useState('');
+  const [nextPurchase, setNextPurchase] = useState(0);
+  const [lastPurchase, setLastPurchase] = useState(null);
 
-  const handleClick = async (e) => {
-    try {
-      setLoading(true);
+  const currToken = localStorage.getItem('currToken');
+  const itemsCollectionRef = collection(db, 'shopping-list');
 
-      await addDoc(collection(db, 'shopping-list'), {
-        itemName: 'sugar',
-        buyingTime: 'today',
+  const addItems = async () => {
+    const refData = await getDocs(itemsCollectionRef);
+    let tokenList = refData.docs.map(({ id }) => id);
+
+    if (tokenList.includes(currToken)) {
```

Comment:
You don't really need to add logic to check if the token exists, we assume here that we already have a token to work with. 

Once you merge this story with #3 you will be able to remove this  conditional and `tokenList` logic